wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentationFlowActivity.kt

```diff
@@ -0,0 +1,294 @@
+package com.android.identity_credential.wallet.presentation
+
+import androidx.fragment.app.FragmentActivity
+import com.android.identity.android.securearea.AndroidKeystoreKeyUnlockData
+import com.android.identity.android.securearea.AndroidKeystoreSecureArea
+import com.android.identity.android.securearea.UserAuthenticationType
+import com.android.identity.cbor.Cbor
+import com.android.identity.crypto.Algorithm
+import com.android.identity.document.Document
+import com.android.identity.document.NameSpacedData
+import com.android.identity.issuance.DocumentExtensions.documentConfiguration
+import com.android.identity.mdoc.credential.MdocCredential
+import com.android.identity.mdoc.mso.MobileSecurityObjectParser
+import com.android.identity.mdoc.mso.StaticAuthDataParser
+import com.android.identity.mdoc.response.DeviceResponseGenerator
+import com.android.identity.mdoc.response.DocumentGenerator
+import com.android.identity.mdoc.util.MdocUtil
+import com.android.identity.securearea.KeyLockedException
+import com.android.identity.securearea.KeyUnlockData
+import com.android.identity.securearea.software.SoftwareKeyInfo
+import com.android.identity.securearea.software.SoftwareKeyUnlockData
+import com.android.identity.securearea.software.SoftwareSecureArea
+import com.android.identity.util.Constants
+import com.android.identity_credential.wallet.R
+import com.android.identity_credential.wallet.WalletApplication
+import com.android.identity_credential.wallet.presentation.PresentationFlowActivity.PromptUnsuccessfulException
+import com.android.identity_credential.wallet.ui.prompt.biometric.showBiometricPrompt
+import com.android.identity_credential.wallet.ui.prompt.consent.showConsentPrompt
+import com.android.identity_credential.wallet.ui.prompt.passphrase.showPassphrasePrompt
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.datetime.Clock
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * A base Activity that provides functions for initiating an MDL Presentation flow and obtaining the
+ * response bytes to send to the requesting party.
+ *
+ * Subclassing Activities call function [showPresentationFlow] with the request bytes parsed to a
+ * [PresentationRequestData]. Response bytes are (eventually) generated after authentication prompts
+ * validate the user successfully, else an exception is thrown.
+ *
+ * Exceptions are thrown for numerous reasons such as,
+ * - [IllegalStateException] if credentials could not be found in the Document; authentication
+ * type was not specified when showing Biometric prompt; a new secure area has not been handled yet
+ * - [PromptUnsuccessfulException] if a prompt is unsuccessful when user cancels the prompt or is
+ * unable to authenticate.
+ */
+open class PresentationFlowActivity : FragmentActivity() {
```

Comment:
Why are we providing this functionality via a `FragmentActivity` base class instead of just providing a function? I think my preference is that what we'd expose is this:

```
suspend fun mdocPresentationFlow(
  activity: FragmentActivity,
  document: Document,
  documentRequest: DocumentRequest,
  docType : String,
  trustPoint : TrustPoint?,
  sessionTranscript: ByteArray): ByteArray
```

which returns the `DeviceResponse` CBOR. You mention `PromptUnsuccessfulException`... I'm not sure we actually catch that anywhere and I can't think of a good reason why the caller would need it. I would suggest to just remove that, we can always add it back if we need it.